examples/native/helpers/driver.sh

```diff
@@ -0,0 +1,174 @@
+
+function _docker_exec_with_retry() {
+
+    local RETRY_COUNT=5
+
+    local CONTAINER=$1
+    local COMMAND=$2
+    local SUCCESS_MSG=${3:-""}
+
+    local i=0
+    for ((i; i<RETRY_COUNT; i++)); do
+        output=$(docker exec "$CONTAINER" /usr/bin/env bash -c "$COMMAND")
+        if [[ $output == *$SUCCESS_MSG* ]]; then
+            return
+        else
+            sleep 2
+            echo "Retrying..."
+        fi
+    done
+    echo "Max retries reached, $CONTAINER failed"
+
+}
+# Pre-conditions: 
+#   - $NUM_NODES containers running Couchbase Server (uninitialised)/exporter 
+
+# Post-conditions: 
+#   - All CBS/exporter nodes initialised and partitioned as evenly as possible into 
+#     $CLUSTER_NUM clusters, with a rebalance occurring after the last node is added
+#   - $CLUSTER_NUM nodes will be running the Data Service, the rest Index/Query, with quotas
+#     specified by $DATA_ALLOC and $INDEX_ALLOC
+#   - Every cluster registered for monitoring with the cbmultimanager
+function configure_servers() {
+
+    local NUM_NODES=$1
+    local CLUSTER_NUM=$2
+    local SERVER_USER=$3
+    local SERVER_PWD=$4
+    local NODE_RAM=$5
+    local LOAD=$6
+
+    local DATA_ALLOC 
+    local INDEX_ALLOC
+    # Allocate 70% of the specified RAM quota to the service (query has no quota)
+    DATA_ALLOC=$(awk -v n="$NODE_RAM" 'BEGIN {printf "%.0f\n", (n*0.7)}')
+    INDEX_ALLOC=$(awk -v n="$NODE_RAM" 'BEGIN {printf "%.0f\n", (n*0.7)}')
+
+    local temp_dir
+    temp_dir=$(mktemp -d)
+    echo '[]' > "$temp_dir"/targets.json
+
+    local nodes_left=$NUM_NODES
+    local i=0
+    for ((i; i<CLUSTER_NUM; i++)); do
+
+        # Calculate the number of nodes to provision in this cluster
+        local to_provision=$(( nodes_left / (CLUSTER_NUM - i) )) # This is always integer division, Bash does not support decimals
+        local start=$(( NUM_NODES - nodes_left ))
+        
+        # Create and initialize cluster
+        local uid="node$start"
+        local clust_name="Cluster $i"
+        _docker_exec_with_retry "$uid" "/opt/couchbase/bin/couchbase-cli cluster-init -c localhost --cluster-name=\"$clust_name\" \
+            --cluster-username=\"$SERVER_USER\" --cluster-password=\"$SERVER_PWD\" --cluster-ramsize=$DATA_ALLOC \
+            --cluster-index-ramsize=$INDEX_ALLOC --services=data || echo 'failed'" "SUCCESS: "
+        local nodes=(\"node"$start".local:9091\")
+
+        # Load sample buckets and register cluster with CBMM
+        _docker_exec_with_retry "$uid" "curl -fs -X POST -u \"$SERVER_USER\":\"$SERVER_PWD\" \"http://localhost:8091/sampleBuckets/install\" \
+          -d '[\"travel-sample\", \"beer-sample\"]'" "[]"
+        
+        local cmos_cmd="curl -fs -u $CLUSTER_MONITOR_USER:$CLUSTER_MONITOR_PWD -X POST -d '{\"user\":\"$SERVER_USER\",\"password\":\"$SERVER_PWD\", \"host\":\"http://$uid:8091\"}' 'http://localhost:8080/couchbase/api/v1/clusters'"
+        _docker_exec_with_retry "cmos" "$cmos_cmd"
```

Comment:
Would it be possible to split that command up into multiple lines using `\`? Might entail removing the temporary variable

Reply:
Yup got a commit for this locally, went through all and reformatted just making sure I haven't broken anything